app/uk/gov/hmrc/agentoverseasfrontend/controllers/GenericEmailVerificationController.scala

```diff
@@ -0,0 +1,150 @@
+    */
+  def redirectUrlIfLocked(session: S): Call
+
+  /**
+    * User will be sent here in case of unexpected errors during verification
+    */
+  def redirectUrlIfError(session: S): Call
+
+  /**
+    * The URL for the back link displayed on the verification page
+    */
+  def backLinkUrl(session: S): Option[Call]
+
+  /**
+    * User may be sent here if they need to re-enter their email
+    */
+  def enterEmailUrl(session: S): Call
+
+  def verifyEmail: Action[AnyContent] = Action.async { implicit request =>
+    getState.flatMap {
+      case (session, credId) =>
+        val emailToVerify = getEmailToVerify(session)
+        if (isAlreadyVerified(session, emailToVerify)) {
+          Future.successful(Redirect(redirectUrlIfVerified(session)))
+        } else {
+          // Check the status of the email with the email verification service
+          emailVerificationService.checkStatus(credId, emailToVerify).flatMap {
+            case EmailVerificationStatus.Verified =>
+              markEmailAsVerified(session, emailToVerify).map { updatedSession =>
+                Redirect(redirectUrlIfVerified(updatedSession))
+              }
+            // The email is not yet verified. Start the verification journey
+            case EmailVerificationStatus.Unverified =>
+              emailVerificationService
+                .verifyEmail(
+                  credId,
+                  Some(
+                    Email(
+                      address = emailToVerify,
+                      enterUrl = urlFor(enterEmailUrl(session))
+                    )),
+                  continueUrl = urlFor(selfRoute), // when the verification journey is done, this same method will be called again to verify that the email is indeed verified
+                  mBackUrl = backLinkUrl(session).map(urlFor(_)),
+                  accessibilityStatementUrl = accessibilityStatementUrl
+                )
+                .map {
+                  case Some(redirectUri) =>
+                    val url = env.mode match {
+                      case Mode.Dev => emailVerificationFrontendBaseUrl + redirectUri
+                      case _        => redirectUri // we need an absolute uri if in Dev, relative otherwise
+                    }
+                    Redirect(url)
+                  case None => throw new RuntimeException("Could not start email verification journey")
+                }
+            // The email provided was locked out due to too many failed verification attempts
+            case EmailVerificationStatus.Locked =>
+              Future.successful(Redirect(redirectUrlIfLocked(session)))
+            // Any other error
+            case EmailVerificationStatus.Error =>
+              Future.successful(Redirect(redirectUrlIfError(session)))
+          }
+        }
+
+    }
+  }
+
+  private def urlFor(call: Call)(implicit request: RequestHeader): String = env.mode match {
+    case Mode.Dev =>
```

Comment:
I believe services that run from service manager are run in Prod mode in which case we would get a relative url. Alternative is to define the url in app-config-base and keeping application.conf local - wdyt?

Reply:
Can we use a feature flag to enable email-verification, with having it switched off in application.conf but on in app-config-base. That way we can test it in development and other environments. Or we can leave as is and we can wait until the work-around is available.